helper/logging/logging_http_transport.go

```diff
@@ -0,0 +1,247 @@
+package logging
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"net/textproto"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// NewLoggingHTTPTransport creates a wrapper around an *http.RoundTripper,
+// designed to be used for the `Transport` field of http.Client.
+//
+// This logs each pair of HTTP request/response that it handles.
+// The logging is done via `tflog`, that is part of the terraform-plugin-log
+// library, included by this SDK.
+//
+// The request/response is logged via tflog.Debug, using the context.Context
+// attached to the http.Request that the transport receives as input
+// of http.RoundTripper RoundTrip method.
+//
+// It's responsibility of the developer using this transport, to ensure that each
+// http.Request it handles is configured with the SDK-initialized Provider Root Logger
+// context.Context, that it's passed to all resources/data-sources/provider entry-points
+// (i.e. schema.Resource fields like `CreateContext`, `ReadContext`, etc.).
+//
+// This also gives the developer the flexibility to further configure the
+// logging behaviour via the above-mentioned context: please see
+// https://www.terraform.io/plugin/log/writing.
+func NewLoggingHTTPTransport(t http.RoundTripper) *loggingHttpTransport {
```

Comment:
Can you ensure there is a changelog file added for these two new function enhancements and a note about the deprecation of the existing transport? Thanks!

Reply:
yep, now that signatures are agreed, I can do that.